onmt/transforms/tokenize.py

```diff
@@ -340,7 +348,7 @@ def _get_subword_kwargs(self, side='src'):
         kwopts = dict()
         if subword_type == 'bpe':
             kwopts['bpe_model_path'] = subword_model
-            kwopts['bpe_dropout'] = subword_alpha
+            kwopts['bpe_dropout'] = subword_alpha if is_train else 0
         elif subword_type == 'sentencepiece':
             kwopts['sp_model_path'] = subword_model
             kwopts['sp_nbest_size'] = subword_nbest
```

Comment:
You can directly reassign on `subword_alpha` & `subword_nbest` to disable both `bpe_dropout`/sp sampling.